src/server/api/blockallowlist.test.ts

```diff
@@ -0,0 +1,415 @@
+import anyTest, { TestFn } from 'ava'
+import sinon from 'sinon'
+import { spawnTestServer } from '../fixtures/spawnServer.js'
+import { FastifyTypebox } from './index.js'
+import ActivityPubSystem from '../apsystem.js'
+
+interface TestContext {
+  server: FastifyTypebox
+  hasAdminPermissionForRequestStub: sinon.SinonStub
+  hasPermissionActorRequestStub: sinon.SinonStub
+  mockStore: any
+}
+
+const test = anyTest as TestFn<TestContext>
+
+test.beforeEach(async t => {
+  t.context.server = await spawnTestServer()
+
+  // Set up the mockStore
+  t.context.mockStore = {
```

Comment:
Instead of setting up a mock store, try using a regular Store with in-memory storage, then instead of stubbing you could add/remove/list expected data.